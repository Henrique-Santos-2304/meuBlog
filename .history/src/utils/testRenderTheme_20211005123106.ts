import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(<ThemeProvider>{children}</ThemeProvider> );
};
