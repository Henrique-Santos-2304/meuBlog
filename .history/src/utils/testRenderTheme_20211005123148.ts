import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children: React.ReactElement): RenderResult => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider> );
};
