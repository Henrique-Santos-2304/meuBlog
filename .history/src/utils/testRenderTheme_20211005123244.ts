import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider> );
};
