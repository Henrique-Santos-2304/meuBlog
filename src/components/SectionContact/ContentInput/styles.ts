import styled, { css } from "styled-components";
import { FaExclamationCircle } from "react-icons/fa";

export const ContentInputs = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: center;
  padding: 1rem 0.6rem;
  margin: 0.6rem 0;
`;
export const Label = styled.label`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xsmall};
    margin: 0.4rem;
    color: ${theme.colors.primary};

    @media (min-width: ${theme.media.xmedium}) {
      font-size: ${theme.font.sizes.small};
    }
  `}
`;
export const Alert = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    padding: 0.1rem ${theme.spacings.xxsmall};
    color: ${theme.colors.attention};
    font-size: ${theme.font.sizes.xsmall};
    font-weight: 500;
    border-radius: 0.5rem;
    margin-top: 0.2rem;
  `}
`;
export const Icon = styled(FaExclamationCircle)`
  ${({ theme }) => css`
    background: ${theme.colors.secondary};
    border-radius: 50%;
    font-size: ${theme.font.sizes.small};
    font-weight: 400;
    margin-right: 0.4rem;
  `}
`;
