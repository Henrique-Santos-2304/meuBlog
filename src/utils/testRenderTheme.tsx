import { ThemeProvider } from "styled-components";
import { ChakraProvider } from "@chakra-ui/react";

import { render, RenderResult } from "@testing-library/react";
import theme from "styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(
    <ThemeProvider theme={theme}>
      <ChakraProvider> {children}</ChakraProvider>
    </ThemeProvider>
  );
};
