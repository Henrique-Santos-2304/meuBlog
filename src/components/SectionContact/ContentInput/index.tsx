import * as S from "./styles";

type props = {
  children: React.ReactNode;
  error: string | undefined;
  typeLabel: string;
  txtLabel: string;
};
const ContentInput = ({ children, error, typeLabel, txtLabel }: props) => (
  <S.ContentInputs>
    <S.Label htmlFor={typeLabel}>{txtLabel}</S.Label>
    {children}
    {error && (
      <S.Alert role="alert">
        <S.Icon></S.Icon>
        <p>{error}</p>
      </S.Alert>
    )}
  </S.ContentInputs>
);

export default ContentInput;
