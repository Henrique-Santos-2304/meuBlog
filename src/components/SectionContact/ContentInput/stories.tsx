import { Story, Meta } from "@storybook/react";
import ContentInput from ".";

export default {
  title: "ContentInput",
  component: ContentInput,
} as Meta;
const args = {
  error: "deu erro",
  typeLabel: "name",
  txtLabel: "Nome",
};
export const Basic: Story = () => (
  <ContentInput {...args}>
    <p>Sou Children</p>
  </ContentInput>
);

export const Default: Story = () => (
  <ContentInput {...args}>
    <p>Sou Children</p>
  </ContentInput>
);
