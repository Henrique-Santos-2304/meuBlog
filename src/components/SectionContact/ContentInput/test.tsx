import ContentInput from ".";
import { screen } from "@testing-library/react";
import { renderTheme } from "utils/testRenderTheme";

const mock = {
  error: "deu erro",
  typeLabel: "name",
  txtLabel: "Nome",
};
describe("<ContentInput />", () => {
  it("should render ", () => {
    const { container } = renderTheme(
      <ContentInput {...mock}>
        <p>Sou Children</p>
      </ContentInput>
    );
    expect(container).toMatchSnapshot();
  });
  it("should render ", () => {
    renderTheme(
      <ContentInput {...mock} error="">
        <p>Sou Children</p>
      </ContentInput>
    );
    const error = screen.queryByText(/deu erro/i);
    expect(error).not.toBeInTheDocument();
  });
});
