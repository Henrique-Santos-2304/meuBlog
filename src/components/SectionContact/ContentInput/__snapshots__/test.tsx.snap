// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContentInput /> should render  1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1rem 0.6rem;
  margin: 0.6rem 0;
}

.c1 {
  font-size: 1.2rem;
  margin: 0.4rem;
  color: #f8f8f8;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.1rem 0.8rem;
  color: #D3D542;
  font-size: 1.2rem;
  font-weight: 500;
  border-radius: 0.5rem;
  margin-top: 0.2rem;
}

.c3 {
  background: #1e1e1e;
  border-radius: 50%;
  font-size: 1.4rem;
  font-weight: 400;
  margin-right: 0.4rem;
}

@media (min-width:768px) {
  .c1 {
    font-size: 1.4rem;
  }
}

<div>
  <div
    class="c0"
  >
    <label
      class="c1"
      for="name"
    >
      Nome
    </label>
    <p>
      Sou Children
    </p>
    <div
      class="c2"
      role="alert"
    >
      <svg
        class="c3"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
        />
      </svg>
      <p>
        deu erro
      </p>
    </div>
  </div>
</div>
`;
